### ISSUE TITLE
waitForProperty is undefined?

### ISSUE THREAD

**Issue by BryanCrotaz** [machty/ember-concurrency#207]
Using `ember-concurrency 0.8.15` (latest)

```
import { task, waitForProperty } from 'ember-concurrency';


	_validityChanged: task(function*() {
		while (true) {
			yield waitForProperty(this, "validations.isValid", value => true);
			this._publishValidity();
		}
	}).on('init'),
```

Error:

```
rsvp.js:27 Uncaught TypeError: (0 , _emberConcurrency.waitForProperty) is not a function
    at Class.<anonymous> (validated-component.js:27)
    at Generator.next (<anonymous>)
    at Class._resumeGenerator (-task-instance.js:431)
    at Class._handleResolvedContinueValue (-task-instance.js:586)
    at Class._proceed (-task-instance.js:550)
    at Class._scheduleProceed (-task-instance.js:538)
    at Class._start (-task-instance.js:242)
    at Class._startTaskInstance (-scheduler.js:104)
    at Class._flushQueues (-scheduler.js:80)
    at Class.schedule (-scheduler.js:62)
```

**Comment by BryanCrotaz** [comment]
I can't find the export in the `app/` folder?

**Comment by BryanCrotaz** [comment]
Interestingly I would have expected
```
yield waitForProperty(this, "validations.isValid", value => true);
```
to complete whenever `validations.isValid` changes. Instead it completes continuously and hangs the browser.

**Comment by machty** [comment]
@BryanCrotaz are you 100% sure you've fully upgraded to the latest version? Seems like this [twiddle](https://ember-twiddle.com/b2b0c016f4df24261381487b60c707f3?numColumns=2&openFiles=controllers.application.js%2Ctemplates.application.hbs) works and gives a quick demo of waitForProperty.

Not sure what you're trying to do in your code, but passing `(value) => true` means it your callback will return true immediately and the task will continue running, and your while loop will continue running infinitely.

**Comment by BryanCrotaz** [comment]
@machty tried using 0.8.15 and latest master from github url. In both cases I've had to copy `utils/util.js` and `utils/-wait-for.js` into my app.

**Comment by machty** [comment]
@BryanCrotaz after running `npm/yarn install` what is the version listed in `node_modules/ember-concurrency/package.json`?

**Comment by tombakerjr** [comment]
@BryanCrotaz I'm having the same issue, and it appears to be due to an older version of Ember Power Select, which specifies an older version of Ember Concurrency. Even though I've defined Ember Concurrency as a top-level dependency, and `node_modules/ember-concurrency/package.json` shows `0.8.15` as the version, I'm unable to use `waitForProperty`.

I'm going to work on updating some related dependencies to see if that sorts it out and will report back here if I remember!

**Comment by machty** [comment]
Wow that's super gross :(

**Comment by BryanCrotaz** [comment]
@cibernox are you able to update the version of ember-concurrency in power-select?

**Comment by tombakerjr** [comment]
The latest 1.x version (not sure how far back it goes) of Ember Power Select specifies `"ember-concurrency": "^0.8.12"`, which should resolve to `0.8.15`. I happened to be way back on Ember Power Select 1.2.x, which wanted a 0.7.x version of Ember Concurrency.

**Comment by BryanCrotaz** [comment]
Ok that’s not the problem then

**Comment by rwjblue** [comment]
What does `npm ls ember-concurrency` show?

**Comment by machty** [comment]
TL;DR yarn doesn't aggressively dedupe your project's dependencies, and other dependencies' dependencies get checked into yarn.lock, and ember-cli merges ALL of the detected e-c versions to the broccoli tree in a non deterministic order, so there's no guarantee which version of e-c you're importing into your app. The solution is to `yarn upgrade` every dependency that might be using e-c, or to nuke your node_modules and yarn.lock and do `yarn install`.

I'm sure there's plenty more discussions to be had regarding this ember-cli behavior but this does not appear to be an ember-concurrency-specific bug.

**Comment by bendemboski** [comment]
For posterity if anyone's interested, here's a project that repros the issue: https://github.com/bendemboski/ember-cli-deps

```
/tmp/ember-cli-deps$ yarn list ember-concurrency
yarn list v1.3.2
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ ember-concurrency@0.8.15
└─ ember-zz-bad-addon@0.0.0
   └─ ember-concurrency@0.8.12
```

**Comment by lukemelia** [comment]
side note: ember-cli-dependency-lint can be helpful in flagging this sort of risk

**Comment by BryanCrotaz** [comment]
@lukemelia wow that’s an excellent addon. Given the nasty side effects of this largely hidden problem, maybe we should consider making it a default part of ember-cli?

**Comment by machty** [comment]
@BryanCrotaz i believe that idea is on the cli team's agenda
